src/UI/LevelInfoUI.cpp

```diff
@@ -88,12 +93,51 @@ namespace LevelInfoUI {
             self->beatmapCharacteristicSegmentedControlController == NULL ||
             self->beatmapCharacteristicSegmentedControlController->selectedBeatmapCharacteristic == NULL) return;
         if (starsLabel == NULL) {
+
+            ///////////////////////////
+            // Skill Triangle
+            ///////////////////////////
+
+            // Create Modal
+            skillTriangleContainer = QuestUI::BeatSaberUI::CreateModal(self->get_transform(), {40,40}, nullptr, true);
+
+            // Create Actual Triangle Image
+            auto skillTriangleImage = QuestUI::BeatSaberUI::CreateImage(skillTriangleContainer->get_transform(), BundleLoader::bundle->beatLeaderLogoGradient, {0, 0}, {35, 35});
+            skillTriangleMat = UnityEngine::Material::Instantiate(BundleLoader::bundle->skillTriangleMaterial);
+            skillTriangleImage->set_material(skillTriangleMat.ptr());
+            int normalizedValuesPropertyId = UnityEngine::Shader::PropertyToID("_Normalized");
+
+            // Create Star Value Labels for Triangle
+            auto techLabel = QuestUI::BeatSaberUI::CreateText(skillTriangleContainer->get_transform(), "Tech - ", {12, 12});
+            auto accLabel = QuestUI::BeatSaberUI::CreateText(skillTriangleContainer->get_transform(), "Acc - ", {34, 12});
+            auto passLabel = QuestUI::BeatSaberUI::CreateText(skillTriangleContainer->get_transform(), "Pass - ", {23, -17});
+
+            // OnClick Function to open the SkillTriangle
+            auto openSkillTriangle = [techLabel, accLabel, passLabel, normalizedValuesPropertyId](){
+                int mapType = currentlySelectedDiff.status;
+                if(mapType != 0 && mapType != 4 && mapType != 5) {
+                    techLabel->SetText("Tech - " + to_string_wprecision(currentlySelectedDiff.techRating, 2));
+                    accLabel->SetText("Acc - " + to_string_wprecision(currentlySelectedDiff.accRating, 2));
+                    passLabel->SetText("Pass - " + to_string_wprecision(currentlySelectedDiff.passRating, 2));
+                    skillTriangleMat->SetVector(normalizedValuesPropertyId, {
+                        clamp(currentlySelectedDiff.techRating / 15.0f, 0.0f, 1.0f),
+                        clamp(currentlySelectedDiff.accRating / 15.0f, 0.0f, 1.0f),
+                        clamp(currentlySelectedDiff.passRating / 15.0f, 0.0f, 1.0f),
+                        0.0f
+                    });
+                    skillTriangleContainer->Show(true, true, NULL);
```

Comment:
We need to automatically hide modal on exit from the song list or ScoreSaber toggle, as it gets stuck there
![image](https://user-images.githubusercontent.com/8793817/232279907-17b529d3-f8fa-4eaa-a4b1-db07accc0f3f.png)

Reply:
Should work now :) Can you test it again? (Especially ScoreSaber Toggle as i dont have ScoreSaber installed)